Add timestamps and tid to logging output.

dhylands opened the pull request:
This also sets the default logging level to INFO.

You can still override this by running foxbox using:
```
RUST_LOG=foxbox=debug cargo run
```

dhylands commented:
r? @fabrice

dhylands commented:
r? @fabricedesre (gotta spell it right)

fabricedesre commented:
Hm, it's sad that nix doesn't provide gettid() since we already depend on it. Maybe you can send them a PR (https://github.com/nix-rust/nix) instead of adding this custom sub-crate?

dhylands commented:
> Hm, it's sad that nix doesn't provide gettid() since we already depend on it. Maybe you can send them a PR (https://github.com/nix-rust/nix) instead of adding this custom sub-crate?

I created https://github.com/nix-rust/nix/pull/293 let's see if I can get that landed.

dhylands commented:
gettid is now in nix (currently just on github, it hasn't made it to a crate yet, but will show up in nix 0.5.1)

So I rebased to the latest master and dropped the sub-crate. Do we need foxbox to build on non-linux platforms? If yes, I can add support so that the gettid field isn't shown on non-linux platforms.

fabricedesre commented:
We have people building on mac, so hiding the thread id on non-linux is a must, yes.

dhylands commented:
Updated so that it only tries to call gettid if building for linux or android (the same platforms which nix::unistd::gettid is available).
